streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -486,7 +486,7 @@
                     Importance.HIGH,
                     CommonClientConfigs.BOOTSTRAP_SERVERS_DOC)
             .define(REPLICATION_FACTOR_CONFIG,
-                    Type.INT,
+                    Type.SHORT,
```

Comment:
Do we have to include this change in this PR? I know this is a small change, but it'd be nice to keep this PR focused only on the Connect-specific changes proposed in KIP-465.